src/components/InteractionLayer/index.js

```diff
@@ -419,24 +425,58 @@ class InteractionLayer extends React.Component {
         newPoints.push({ id: s.id });
       }
     });
+    return newPoints;
+  };
 
-    if (ruler && ruler.visible) {
-      this.setState({ points: newPoints });
+  setRulerPosition = timestamp => {
+    if (!timestamp) {
+      this.setState({
+        points: [],
+        touchX: null,
+        touchY: null,
+      });
+      return;
     }
+    const { xScalerFactory, width, timeSubDomain } = this.props;
+    const xScale = xScalerFactory(timeSubDomain, width);
+    const xpos = xScale(timestamp);
+    this.setRulerPoints(xpos);
+    this.setState({
+      touchX: xpos,
+    });
+  };
+
+  setRulerPoints = xpos => {
+    const { ruler } = this.props;
+    if (!ruler || !ruler.visible) {
+      return [];
+    }
+    const rulerPoints = this.getRulerPoints(xpos);
+    this.setState({ points: rulerPoints });
 
+    return rulerPoints;
+  };
+
+  setArea = (xpos, ypos) => {
     const { area } = this.state;
-    if (area) {
-      const output = this.getDataForCoordinate(xpos, ypos, true);
-      this.setState({
-        area: {
-          ...area,
-          end: output,
-        },
-      });
+    if (!area) {
+      return;
     }
+    const output = this.getDataForCoordinate(xpos, ypos, true);
+    this.setState({
+      area: {
+        ...area,
+        end: output,
+      },
+    });
+  };
 
+  processMouseMove = (xpos, ypos, e = null) => {
+    const rulerPoints = this.setRulerPoints(xpos);
+    this.setArea(xpos, ypos);
+    const { onMouseMove } = this.props;
     if (onMouseMove) {
-      onMouseMove({ points: newPoints, xpos, ypos });
+      onMouseMove({ points: rulerPoints, xpos, ypos, e });
```

Comment:
Talking of API, I would expect `event` to be a first parameter, and then an object of others